src/api/accounts.rs

```diff
@@ -92,72 +47,132 @@ impl<T: Transport> Accounts<T> {
-        let address = signing::recover(message_hash.as_bytes(), &signature, recovery_id)?;
-        Ok(address)
+        /// Sign arbitrary string data.
+        ///
+        /// The data is UTF-8 encoded and enveloped the same way as with
+        /// `hash_message`. The returned signed data's signature is in 'Electrum'
+        /// notation, that is the recovery value `v` is either `27` or `28` (as
+        /// opposed to the standard notation where `v` is either `0` or `1`). This
+        /// is important to consider when using this signature with other crates.
+        pub fn sign<S>(&self, message: S, key: impl signing::Key) -> SignedData
+        where
+            S: AsRef<[u8]>,
+        {
+            let message = message.as_ref();
+            let message_hash = self.hash_message(message);
+
+            let signature = key
+                .sign(&message_hash.as_bytes(), None)
+                .expect("hash is non-zero 32-bytes; qed");
+            let v = signature
+                .v
+                .try_into()
+                .expect("signature recovery in electrum notation always fits in a u8");
+
+            let signature_bytes = Bytes({
+                let mut bytes = Vec::with_capacity(65);
+                bytes.extend_from_slice(signature.r.as_bytes());
+                bytes.extend_from_slice(signature.s.as_bytes());
+                bytes.push(v);
+                bytes
+            });
+
+            // We perform this allocation only after all previous fallible actions have completed successfully.
+            let message = message.to_owned();
+
+            SignedData {
+                message,
+                message_hash,
+                v,
+                r: signature.r,
+                s: signature.s,
+                signature: signature_bytes,
+            }
+        }
+
+        /// Recovers the Ethereum address which was used to sign the given data.
+        ///
+        /// Recovery signature data uses 'Electrum' notation, this means the `v`
+        /// value is expected to be either `27` or `28`.
+        pub fn recover<R>(&self, recovery: R) -> error::Result<Address>
+        where
+            R: Into<Recovery>,
+        {
+            let recovery = recovery.into();
+            let message_hash = match recovery.message {
+                RecoveryMessage::Data(ref message) => self.hash_message(message),
+                RecoveryMessage::Hash(hash) => hash,
+            };
+            let (signature, recovery_id) = recovery
+                .as_signature()
+                .ok_or_else(|| error::Error::Recovery(signing::RecoveryError::InvalidSignature))?;
+            let address = signing::recover(message_hash.as_bytes(), &signature, recovery_id)?;
+            Ok(address)
+        }
     }
 }
 
 /// A transaction used for RLP encoding, hashing and signing.
+#[cfg(feature = "signing")]
 struct Transaction {
```

Comment:
If `Transaction` is only used in case of signing, why not move it to the `mod signing` along with all the use statements? The goal is to minimize the number of `#[cfg(feature = "signing")]` :)